route-rs-runtime/src/link/blackhole_link.rs

```diff
@@ -1,28 +1,55 @@
-use crate::link::PacketStream;
+use crate::link::{Link, PacketStream, TokioRunnable};
 use futures::{Async, Future, Poll};
 
 /// Link that drops all packets ingressed.
+#[derive(Default)]
 pub struct BlackHoleLink<Packet: Sized> {
-    input_stream: PacketStream<Packet>,
+    hole: Option<BlackHole<Packet>>,
 }
 
-impl<Packet: Sized> BlackHoleLink<Packet> {
-    pub fn new(input_stream: PacketStream<Packet>) -> Self {
-        BlackHoleLink { input_stream }
+impl<Packet> BlackHoleLink<Packet> {
+    pub fn new() -> Self {
+        BlackHoleLink { hole: None }
     }
 }
 
-impl<Packet: Sized> Future for BlackHoleLink<Packet> {
+impl<Packet: Sized + 'static> Link<Packet, ()> for BlackHoleLink<Packet> {
+    fn ingressors(&self, ingress_streams: Vec<PacketStream<Packet>>) -> Self {
+        BlackHoleLink {
+            hole: Some(BlackHole::new(ingress_streams)),
+        }
+    }
+
+    fn build_link(self) -> (Vec<TokioRunnable>, Vec<PacketStream<()>>) {
+        if self.hole.is_none() {
+            panic!("Cannot build link! Missing ingress streams")
+        }
+
+        (vec![Box::new(self.hole.unwrap())], vec![])
+    }
+}
+
+pub struct BlackHole<Packet> {
```

Comment:
Potential Nit: Should this be called BlackHoleIngressor? It is awkward naming here, but in all the other links the thing that implements the `TokioRunnable` should be the Ingressor, and the thing that Impls `Stream` would be the egressor?

Reply:
Nit followup: Implements is awkward phrasing, the thing that is a TokioRunnable object IE, the struct that implements Future, is the ingressor in all other cases.  I think we can leave this as is for now, and revisit later once the other implementations shake out. The names my clarify. I think leaving it blackhole is fine in either case. Just wanted to make the suggestion.

Reply:
Yep, we should revisit namings when more of the Links have been migrated. I went back and forth between a few ideas in the next commit(s).